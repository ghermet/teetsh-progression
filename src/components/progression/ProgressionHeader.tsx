import type { ProgressionData } from '../../types/progression.types';

/**
 * Props for the ProgressionHeader component
 */
type ProgressionHeaderProps = Pick<ProgressionData, 'name' | 'niveau'> & {
  /** Optional short description of the progression */
  shortDescription?: string;
  /** Optional date when the progression was created */
  date?: string;
};

/**
 * Header component that displays progression metadata including name, description, level, and year.
 *
 * Renders a styled header section with the progression title, description, educational level,
 * and the year extracted from the creation date.
 *
 * @param props - Header content and metadata
 * @param props.name - The progression title/name
 * @param props.shortDescription - Optional brief description
 * @param props.niveau - Educational level (e.g., "CP", "CE1")
 * @param props.date - Optional creation date (year will be extracted)
 * @returns Styled header section with progression metadata
 */
export function ProgressionHeader({
  name,
  shortDescription,
  niveau,
  date,
}: ProgressionHeaderProps) {
  return (
    <header className="bg-slate-100 px-4 sm:px-6 py-6 sm:py-8 border-b border-slate-200">
      <div className="flex flex-col sm:flex-row sm:items-center sm:justify-between gap-4 sm:gap-0">
        <div className="flex-1 min-w-0">
          <div className="flex items-center justify-between flex-wrap gap-1">
            <h1 className="text-xl md:text-2xl lg:text-3xl text-slate-800 font-medium">
              {name}
            </h1>
            <div className="flex items-baseline gap-2 text-slate-600 text-sm sm:text-base flex-shrink-0">
              <strong className="text-slate-800">{niveau}</strong>
              {date && (
                <>
                  <span className="hidden sm:inline">â€¢</span>
                  <span className="text-slate-700">
                    {new Date(date).getFullYear()}
                  </span>
                </>
              )}
            </div>
          </div>
          <p className="text-sm md:text-base lg:text-lg text-slate-600 mt-1">
            {shortDescription}
          </p>
        </div>
      </div>
    </header>
  );
}
