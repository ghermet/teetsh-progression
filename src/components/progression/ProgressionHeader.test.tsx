/** Generated with CLAUDE CODE */

import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { ProgressionHeader } from './ProgressionHeader';

describe('ProgressionHeader', () => {
  const defaultProps = {
    name: 'Test Timeline',
    shortDescription: 'A test timeline description',
    niveau: 'CP',
    date: '2024-01-15',
  };

  it('renders the timeline name', () => {
    render(<ProgressionHeader {...defaultProps} />);
    expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent(
      'Test Timeline'
    );
  });

  it('renders the short description when provided', () => {
    render(<ProgressionHeader {...defaultProps} />);
    expect(screen.getByText('A test timeline description')).toBeInTheDocument();
  });

  it('does not render short description when not provided', () => {
    const propsWithoutDescription = {
      name: 'Test Timeline',
      niveau: 'CP',
      date: '2024-01-15',
    };
    render(<ProgressionHeader {...propsWithoutDescription} />);
    expect(
      screen.queryByText('A test timeline description')
    ).not.toBeInTheDocument();
  });

  it('renders the niveau', () => {
    render(<ProgressionHeader {...defaultProps} />);
    expect(screen.getByText('CP')).toBeInTheDocument();
  });

  it('renders the year from date when provided', () => {
    render(<ProgressionHeader {...defaultProps} />);
    expect(screen.getByText('2024')).toBeInTheDocument();
  });

  it('does not render year when date is not provided', () => {
    const propsWithoutDate = {
      name: 'Test Timeline',
      niveau: 'CP',
      shortDescription: 'A test timeline description',
    };
    render(<ProgressionHeader {...propsWithoutDate} />);
    expect(screen.queryByText('2024')).not.toBeInTheDocument();
    expect(screen.queryByText('•')).not.toBeInTheDocument();
  });

  it('renders separator bullet when both niveau and date are provided', () => {
    render(<ProgressionHeader {...defaultProps} />);
    expect(screen.getByText('•')).toBeInTheDocument();
  });

  it('has correct CSS classes for styling', () => {
    const { container } = render(<ProgressionHeader {...defaultProps} />);
    const header = container.querySelector('header');
    expect(header).toHaveClass(
      'bg-slate-100',
      'px-4',
      'sm:px-6',
      'py-6',
      'sm:py-8',
      'border-b',
      'border-slate-200'
    );
  });

  it('matches snapshot', () => {
    const { container } = render(<ProgressionHeader {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
