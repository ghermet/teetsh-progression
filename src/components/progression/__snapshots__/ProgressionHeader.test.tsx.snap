// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProgressionHeader > matches snapshot 1`] = `
<div>
  <header
    class="bg-slate-100 px-4 sm:px-6 py-6 sm:py-8 border-b border-slate-200"
  >
    <div
      class="flex flex-col sm:flex-row sm:items-center sm:justify-between gap-4 sm:gap-0"
    >
      <div
        class="flex-1 min-w-0"
      >
        <div
          class="flex items-center justify-between flex-wrap gap-1"
        >
          <h1
            class="text-xl md:text-2xl lg:text-3xl text-slate-800 font-medium"
          >
            Test Timeline
          </h1>
          <div
            class="flex items-baseline gap-2 text-slate-600 text-sm sm:text-base flex-shrink-0"
          >
            <strong
              class="text-slate-800"
            >
              CP
            </strong>
            <span
              class="hidden sm:inline"
            >
              â€¢
            </span>
            <span
              class="text-slate-700"
            >
              2024
            </span>
          </div>
        </div>
        <p
          class="text-sm md:text-base lg:text-lg text-slate-600 mt-1"
        >
          A test timeline description
        </p>
      </div>
    </div>
  </header>
</div>
`;
